R/fread.R

```diff
@@ -79,8 +73,144 @@ fread <- function(input="",file,sep="auto",sep2="auto",dec=".",quote="\"",nrows=
     }
   }
   stopifnot(length(skip)==1L, !is.na(skip), is.character(skip) || is.numeric(skip))
-  if (skip=="__auto__") skip=-1L   # skip="string" so long as "string" is not "__auto__". Best conveys to user something is automatic there (than -1 or NA).
+  if (skip=="__auto__") {
+    if (has_yaml) {
+      skip=0L
+    } else {
+      skip=-1L   # skip="string" so long as "string" is not "__auto__". Best conveys to user something is automatic there (than -1 or NA).
+    }
+  }
   if (is.double(skip)) skip = as.integer(skip)
+  if (has_yaml) {
+    if (!require('yaml'))
```

Comment:
`requireNamespace('yaml', quietly = TRUE)` I think is preferred

Reply:
Yes indeed, travis gave me the same warning